client/engine/engine.go

```diff
@@ -251,57 +250,20 @@ func (e *Engine) constructObjectiveFromMessage(message protocols.Message) (proto
 	initialState := message.SignedStates[0].State()
 
 	switch {
-	case strings.Contains(string(message.ObjectiveId), `DirectFund`):
+	case directfund.IsDirectFundObjective(message.ObjectiveId):
 
-		if initialState.TurnNum != 0 {
-			return directfund.Objective{}, errors.New("cannot construct direct fund objective without prefund state")
-		}
 		return directfund.NewObjective(
 			true, // TODO ensure objective in only approved if the application has given permission somehow
 			initialState,
 			*e.store.GetAddress(),
 		)
```

Comment:
Nit: it might be nice for this branch of the switch statement to also call into a `directfund.ConstructObjectiveFromMessage` function?

Reply:
Good point! [fc3e6e9](https://github.com/statechannels/go-nitro/pull/298/commits/fc3e6e997e83413faafa358c21b14bd5cef18500)